### DISCUSSION TITLE
How to launch server for playing with Hamachi

### DISCUSSION THREAD

**Discussion by sannoization** [itzg/docker-minecraft-server#1417] [Q&A]
Как открыть сервер для LAN, чтобы к нему можно было подключиться через сеть в Hamachi с другого компа?
Я использую docker-compose для того, чтобы поднять сервер и хотел бы какое-то решение для docker-compose.

How to open server for LAN? 
I'm using docker-compose to setup server and would like to get solution for docker-compose.

**Comment by d3adb5** [comment]
This really depends on how Docker functions on your machine. When I expose ports in my Compose file, Docker binds itself on all interfaces (`0.0.0.0:<port>` + `:::<port>`). For the services I wanted exposed only locally, I had to add the IP address manually: `127.0.0.1:25565:25565`, for example.

If Docker isn't binding on all interfaces by default, making it so the host has port 25565 exposed to the public, you can try specifying `0.0.0.0` on the ports mapping, or the IP for the interface that Hamachi creates. Check the IP addresses for your network interfaces or if the Hamachi UI shows you that information. On most GNU/Linux distributions, you can do so through `ip a`. On Windows, `ipconfig /all` might be what you're after.
